Source/WebCore/page/LocalFrame.cpp

```diff
@@ -1307,11 +1307,8 @@ void LocalFrame::didAccessWindowProxyPropertyViaOpener(WindowProxyProperty prope
     if (!opener())
         return;
 
-    auto openerMainFrame = dynamicDowncast<LocalFrame>(opener()->mainFrame());
-    if (!openerMainFrame)
-        return;
-
-    auto openerMainFrameOrigin = SecurityOriginData::fromFrame(openerMainFrame);
+    auto* openerMainFrame = dynamicDowncast<LocalFrame>(opener()->mainFrame());
+    auto openerMainFrameOrigin = openerMainFrame ? SecurityOriginData::fromFrame(openerMainFrame) : opener()->page()->mainFrameOrigin().data();
```

Comment:
It may be worth null checking the opener's page.  We may also be able to always use mainFrameOrigin() instead of checking if the main frame is local and using it if it is to access what should be the same thing.